parsec/core/mountpoint/manager.py

```diff
@@ -134,6 +132,47 @@ def get_path_in_mountpoint(
         await self._mountpoint_tasks[(workspace_id, timestamp)].cancel_and_join()
         del self._mountpoint_tasks[(workspace_id, timestamp)]
 
+    async def remount_workspace_new_timestamp(
+        self, workspace_id: EntryID, original_timestamp: Pendulum, target_timestamp: Pendulum
+    ) -> PurePath:
+        # TODO : use different workspaces for temp mount
+        if original_timestamp == target_timestamp:
+            return
+        if original_timestamp is None:
+            workspace = self._get_workspace(workspace_id)
+        else:
+            try:
+                workspace = self._get_workspace_timestamped(workspace_id, original_timestamp)
+            except MountpointNotMounted:
+                workspace = self._get_workspace(workspace_id)
+        try:
+            new_workspace = await workspace.to_timestamped(target_timestamp)
+        except FSWorkspaceTimestampedTooEarly as exc:
+            raise MountpointConfigurationWorkspaceFSTimestampedError(
+                f"Workspace `{workspace_id}` didn't exist at `{target_timestamp}`",
+                workspace_id,
+                target_timestamp,
+                workspace.get_workspace_name(),
+            ) from exc
+        self._timestamped_workspacefs[(workspace_id, target_timestamp)] = new_workspace
+
+        curried_runner = partial(
+            self._runner,
+            new_workspace,
+            self.base_mountpoint_path,
+            config=self.config,
+            event_bus=self.event_bus,
+        )
+        runner_task = await start_task(self._nursery, curried_runner)
```

Comment:
Is really necessary to duplicate the runner logic? I felt it would be much simpler to implement a `_load_workspace_timestramped` async method and let `mount_workspace` do its job.

Reply:
Actually I would prefer to refactor the whole class in a new PR, wouldn't it be better?